src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
+
+    Actor_SetScale(&this->dyna.actor, sSpawnData[(this->dyna.actor.params >> 8) & 1].scale);
+    this->actionFunc_168 = func_80BA2218;
+    Actor_SetFocus(&this->dyna.actor, sSpawnData[(this->dyna.actor.params >> 8) & 1].focus);
+
+    if (((this->dyna.actor.params >> 0xF) & 1)) {
+        func_80BA24E8(this);
+        if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) {
+            this->dyna.actor.draw = NULL;
+        }
+        DynaPolyActor_Init(&this->dyna, 0);
+        CollisionHeader_GetVirtual(&D_06000B30, &collisionHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, collisionHeader);
+    } else {
+        ObjWarp2block_SetInactive(this);
+    }
+
+    osSyncPrintf((const char *) "時のブロック(ワープ２) (<arg> %04xH <type> color:%d range:%d)\n",
+    	this->dyna.actor.params & 0xFFFF,
+    	this->dyna.actor.home.rot.z & 7,
+    	((s32) this->dyna.actor.params >> 0xB) & 7);
+}
+
+void ObjWarp2block_Destroy(Actor* thisx, GlobalContext *globalCtx) {
+    ObjWarp2block* this = THIS;
+    if ((this->dyna.actor.params >> 0xF) & 1) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void ObjWarp2block_SetInactive(ObjWarp2block *this) {
+    this->actionFunc_164 = ObjWarp2block_DoNothing;
+    this->dyna.actor.draw = NULL;
+}
+
+void ObjWarp2block_DoNothing(ObjWarp2block *this, GlobalContext *globalCtx) {}
+
+void func_80BA24E8(ObjWarp2block *this) {
+    this->actionFunc_164 = func_80BA24F8;
+}
+
+void func_80BA24F8(ObjWarp2block *this, GlobalContext *globalCtx) {
+    Actor *current;
+
+    current = globalCtx->actorCtx.actorLists[7].head;
+
+    while (current != NULL) {
+        if (current->id == 0x1D6) {
+            if (((current->params >> 0xF) & 1) == 0) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA1ECC.s")
+                if ((this->dyna.actor.params & 0x3F) == (current->params & 0x3F)) {
+                    this->dyna.actor.child = current;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA2048.s")
+                    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) {
+                        ObjWarp2block_SwapWithChild(this, globalCtx);
+                        this->dyna.actor.draw = ObjWarp2block_Draw;
+                    }
+                    func_80BA2600(this);
+                    return;
+                }
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA2218.s")
+        current = current->next;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA228C.s")
+    this->unk_174 = this->unk_174 + 1;
+    if (this->unk_174 >= 0x3D) {
+        osSyncPrintf("\x1b[41;37m");
+        osSyncPrintf("Error : 時のブロック(ワープ２)が対でセットされていません(%s %d)\n", "../z_obj_warp2block.c", 0x1F9);
+        osSyncPrintf("\x1b[m");
```

Comment:
`"\x1b"` is the sign of a vt macro. There's a script for that in tools. Also, line numbers in decimal.